llm-examples/mpt/mpt-7b/03_fine_tune_deepspeed.py

```diff
@@ -0,0 +1,185 @@
+# Databricks notebook source
+# MAGIC %md
+# MAGIC
+# MAGIC # Fine tune MPT-7b with deepspeed
+# MAGIC
+# MAGIC This is to fine-tune [MPT-7b](https://huggingface.co/mosaicml/mpt-7b) models on the [databricks-dolly-15k ](https://huggingface.co/datasets/databricks/databricks-dolly-15k) dataset. The MPT models are [Apache 2.0 licensed](https://huggingface.co/mosaicml/mpt-7b). sciq is under CC BY-SA 3.0 license.
+# MAGIC
+# MAGIC Environment for this notebook:
+# MAGIC - Runtime: 13.2 GPU ML Runtime
+# MAGIC - Instance: `g5.12xlarge` on AWS with 4 A10 GPUs.
+# MAGIC
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC Install the missing libraries
+
+# COMMAND ----------
+
+# Skip this step if running on Databricks runtime 13.2 GPU and above.
+!wget -O /local_disk0/tmp/libcusparse-dev-11-7_11.7.3.50-1_amd64.deb https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64/libcusparse-dev-11-7_11.7.3.50-1_amd64.deb && \
+  dpkg -i /local_disk0/tmp/libcusparse-dev-11-7_11.7.3.50-1_amd64.deb && \
+  wget -O /local_disk0/tmp/libcublas-dev-11-7_11.10.1.25-1_amd64.deb https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64/libcublas-dev-11-7_11.10.1.25-1_amd64.deb && \
+  dpkg -i /local_disk0/tmp/libcublas-dev-11-7_11.10.1.25-1_amd64.deb && \
+  wget -O /local_disk0/tmp/libcusolver-dev-11-7_11.4.0.1-1_amd64.deb https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64/libcusolver-dev-11-7_11.4.0.1-1_amd64.deb && \
+  dpkg -i /local_disk0/tmp/libcusolver-dev-11-7_11.4.0.1-1_amd64.deb && \
+  wget -O /local_disk0/tmp/libcurand-11-7_10.2.10.91-1_amd64.deb https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64/libcurand-11-7_10.2.10.91-1_amd64.deb && \
+  dpkg -i /local_disk0/tmp/libcurand-11-7_10.2.10.91-1_amd64.deb
+
+# COMMAND ----------
+
+# MAGIC %pip install ninja
+# MAGIC %pip install einops==0.5.0 flash-attn==v1.0.3.post0
+# MAGIC %pip install xentropy-cuda-lib@git+https://github.com/HazyResearch/flash-attention.git@v1.0.3#subdirectory=csrc/xentropy
+# MAGIC %pip install triton-pre-mlir@git+https://github.com/vchiley/triton.git@triton_pre_mlir_sm90#subdirectory=python
+# MAGIC %pip install deepspeed
+# MAGIC %pip install xformers torch==2.0.1 sentencepiece==0.1.97
+
+# COMMAND ----------
+
+dbutils.library.restartPython()
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Fine tune the model with `deepspeed`
+# MAGIC
+# MAGIC The fine tune logic is written in `scripts/fine_tune_deepspeed.py`.
+
```

Comment:
Maybe mention the dataset used for fine-tuning here, which is databricks-dolly-15k.